File: app/src/main/java/com/kjipo/eartraining/midistream/MidiStream.java
package com.kjipo.eartraining.midistream;

import android.content.Context;

import com.kjipo.eartraining.midi.MidiMessages;
import com.kjipo.eartraining.midi.MidiPlayerInterface;

import java.io.IOException;


public class MidiStream implements MidiPlayerInterface {
    private Synthesizer synthesizer;


    @Override
    public void noteOn(int pitch) {
        byte command[] = new byte[3];
        command[0] = (byte)MidiMessages.NOTE_ON.message;
        command[1] = (byte)pitch;
        command[2] = 127;
        try {
            // TODO Not necessary to wait here? Put on a different thread
            synthesizer.send(command, 0, 3);
        } catch (IOException e) {
            e.printStackTrace();
        }


    }

    @Override
    public void noteOff(int pitch) {
        byte command[] = new byte[3];
        command[0] = (byte)MidiMessages.NOTE_OFF.message;
        command[1] = (byte)pitch;
        command[2] = 127;
        try {
            // TODO Not necessary to wait here? Put on a different thread
            synthesizer.send(command, 0, 3);
        } catch (IOException e) {
            e.printStackTrace();
        }

    }

    @Override
    public void setup(Context applicationContext) {
        synthesizer = new Synthesizer();
    }

    @Override
    public void start() {
        synthesizer.start();
    }

    @Override
    public void stop() {
        synthesizer.stop();
    }
}

File: app/src/main/java/com/kjipo/eartraining/midistream/MidiStream.kt
package com.kjipo.eartraining.midistream

import com.kjipo.eartraining.midi.MidiMessages
import com.kjipo.eartraining.midi.MidiPlayerInterface
import java.io.IOException


class MidiStream : MidiPlayerInterface {
    private var synthesizer = Synthesizer()


    override fun noteOn(pitch: Int) {
        val command = ByteArray(3)
        command[0] = MidiMessages.NOTE_ON.message.toByte()
        command[1] = pitch.toByte()
        command[2] = 127
        try {
            // TODO Not necessary to wait here? Put on a different thread
            synthesizer.send(command, 0, 3)
        } catch (e: IOException) {
            e.printStackTrace()
        }


    }

    override fun noteOff(pitch: Int) {
        val command = ByteArray(3)
        command[0] = MidiMessages.NOTE_OFF.message.toByte()
        command[1] = pitch.toByte()
        command[2] = 127
        try {
            // TODO Not necessary to wait here? Put on a different thread
            synthesizer.send(command, 0, 3)
        } catch (e: IOException) {
            e.printStackTrace()
        }

    }


    override fun start() {
        synthesizer.start()
    }

    override fun stop() {
        synthesizer.stop()
    }
}
